src/tests/ftest/launch.py

```diff
@@ -20,164 +20,471 @@
 
-    start_time = int(time.time())
-    print("Running: " + test_cmd + "\n\n")
-    subprocess.call(test_cmd, shell=True)
-    end_time = int(time.time())
-    print("Total test run-time in seconds: {}".format(end_time - start_time))
-    # move the job to a path that will be recognizable
-    try:
-        os.makedirs(os.path.join("avocado", os.path.dirname(_file)))
-    except OSError as excepn:
-        if excepn.errno != errno.EEXIST:
-            raise
-    os.rename(os.path.realpath(os.path.join("avocado", "job-results",
-                                            "latest")),
-              os.path.join("avocado", _file))
-    os.remove(os.path.join("avocado", "job-results", "latest"))
-    os.rmdir(os.path.join("avocado", "job-results"))
+    if not tag_filter == "":
+        # Find all the tests that match the tags
+        command = "avocado list --paginator off{} ./ | " \
+                  "grep '^INSTRUMENTED' | awk '{{print $2}}' | " \
+                  "awk -F ':' '{{print $1}}' | uniq".format(tag_filter)
+        test_list.extend(get_output(command).splitlines())
 
+    return tag_filter, test_list
 
-if __name__ == "__main__":
 
-    # not perfect param checking but good enough for now
-    sparse = False
-    addnl_tests = []
-    if len(sys.argv) == 2:
-        if sys.argv[1] == '-h' or sys.argv[1] == '--help':
-            printhelp()
-        test_request = sys.argv[1]
-    elif len(sys.argv) >= 3:
-        if sys.argv[1] == '-s':
-            sparse = True
-            test_request = sys.argv[2]
-        if sparse:
-            remaining_args = 3
+def get_test_files(test_list, machine_mapping, debug):
+    """Get a list of the test scripts to run and their yaml files.
+
+    Args:
+        test_list (list): list of test scripts to run
+        machine_mapping (dict): the mapping of machine replacement keys to
+            their replacement values
+        debug (bool): whether or not to dispaly debug messages
+
+    Returns:
+        list: a list of dictionaries of each test script and yaml file
+
+    """
+    test_files = [{"py": test, "yaml": None} for test in test_list]
+    home_tmp = os.path.join(os.path.expanduser("~"), "tmp")
+    get_output("mkdir -p {}".format(home_tmp))
+    for test_file in test_files:
+        base, _ = os.path.splitext(test_file["py"])
+        file_name = os.path.basename(base)
+        if os.path.exists("{}.yaml".format(base)):
+            # Read the yaml file
+            display("Reading {}".format("{}.yaml".format(base)), debug)
+            with open("{}.yaml".format(base), "r") as yaml_file:
+                data = yaml_file.readlines()
+
+            # Replace the 'boro-[A-Z]' names with the specified machine names.
+            # Exclude any machine name placeholders that do not have a mapping.
+            tmp_yaml = os.path.join(home_tmp, "{}.yaml".format(file_name))
+            display("Creating {}".format(tmp_yaml), debug)
+            with open(tmp_yaml, "w") as yaml_file:
```

Comment:
No, there is no code that cleans up the temporary yaml files, however, since the same name and location is used there should only ever be one temporary file per test.  The original thought was to keep the file around so the "avocado run ... " command could be re-issued by the user, but I don't how much that would get used in practice.

Reply:
I think my default we should clean up after ourselves.  We are in fact running into problems currently because of that lack of clean up (which this patch actually addresses with it's pre-test cleanup feaping creature.  :-)

An option to not clean up, to meet your above goal would be fine I think.